core-services/src/main/java/org/zalando/nakadi/partitioning/PartitionResolver.java

```diff
@@ -56,19 +56,33 @@ public void validate(final EventTypeBase eventType) throws NoSuchPartitionStrate
         }
     }
 
-    public String resolvePartition(final EventType eventType, final JSONObject eventAsJson)
+    public List<String> extractEventPartitionKeys(final EventType eventType, final JSONObject eventAsJson)
             throws PartitioningException {
 
+        return resolveStrategy(eventType).extractEventKeys(eventType, eventAsJson);
+    }
+
+    public String resolvePartition(
```

Comment:
it is strange that `resolvePartition` has `resolveStrategy` inside

Reply:
probably partition resolver depends on partition strategy resolver, but not vice versa

Reply:
probably, to resolve partition it is required to have strategy resolver, maybe pass it as dependency